src/main/java/com/swisscom/aem/tools/jcrhopper/osgi/SampleContributor.java

```diff
@@ -0,0 +1,22 @@
+package com.swisscom.aem.tools.jcrhopper.osgi;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.apache.sling.api.resource.ResourceResolver;
+
+public interface SampleContributor {
+	/**
+	 * Returns a list of sample scripts.
+	 *
+	 * @param resourceResolver Resource resolver of current scope
+	 * @return List of samples with label and configJson
+	 */
+	List<Sample> getSamples(ResourceResolver resourceResolver);
+
+	@RequiredArgsConstructor
+	final class Sample {
+
```

Comment:
Maybe we could add an ID by which we can reference samples?

```suggestion
		private final String id;
```

That would enable a further enhancement (to be implemented later): Execute scripts by referencing a sample instead of submitting the script data:

```bash
curl -X POST -F "_sample=jcrFolder:/apps/jcr-hopper/script-builder/scripts/samples/add-node.json" -F "_commit=$DO_COMMIT" "$AEM_INSTANCE/bin/servlets/jcr-hopper/run"
```

WDYT?

Reply:
but this we are still able to do by providing json path to the script right? Not sure how ID will help in terms of referencing?

Reply:
True, if we only want the samples provided by `JcrFolderSampleContributor` to be accessible this way. But if we want all samples contributed by any `SampleContributor` to be available, something like a generic ID would be required.

But I agree we can postpone this till we actually implement such functionality.

Reply:
True, if we only want the samples provided by `JcrFolderSampleContributor` to be accessible this way. But if we want all samples contributed by any `SampleContributor` to be available, something like a generic ID would be required.

But I agree we can postpone this till we actually implement such functionality.